### ISSUE TITLE
job/PR history assumes presence of started.json

### ISSUE THREAD

**Issue by fejta** [kubernetes/test-infra#12066]
bootstrap only creates a finished.json file when the PR has a merge conflict (no started.json).

We should either update deck to handle non-exceptional situation without warning or else update bootstrap to produce a started.json file in this scenario.

/assign @Katharine 

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76067/pull-kubernetes-node-e2e/1113542968611966982/

```yaml
jsonPayload: {
  component:  "deck"   
  error:  "failed to read started.json: failed to read pr-logs/pull/76067/pull-kubernetes-node-e2e/1113542968611966982/started.json: failed to get reader for GCS object: storage: object doesn't exist"   
  level:  "warning"   
  msg:  "build pr-logs/pull/76067/pull-kubernetes-node-e2e/1113542968611966982/ information incomplete"   
 }
```

Assign back to me if you want me to fix bootstrap.py instead of deck

**Comment by stevekuznetsov** [comment]
:man_facepalming:

**Comment by fejta** [comment]
/remove-lifecycle rotten

Can we fix this? This is the normal state of affairs and therefore should not be considered a warning:


```yaml
{
 insertId:  "1uluqggg7gnqd18"  
 jsonPayload: {
  component:  "deck"   
  error:  "failed to read started.json: failed to read pr-logs/pull/80568/pull-kubernetes-e2e-gce-csi-serial/1158388895834968065/started.json: failed to get reader for GCS object: storage: object doesn't exist"   
  file:  "prow/cmd/deck/pr_history.go:142"   
  func:  "main.getPRBuildData.func1"   
  level:  "warning"   
  msg:  "build pr-logs/pull/80568/pull-kubernetes-e2e-gce-csi-serial/1158388895834968065/ information incomplete"   
 }
 labels: {…}  
 logName:  "projects/k8s-prow/logs/deck"  
 receiveTimestamp:  "2019-08-08T20:41:25.645002916Z"  
 resource: {…}  
 severity:  "ERROR"  
 timestamp:  "2019-08-08T20:41:23Z"  
}
```

**Comment by fejta** [comment]
/kind oncall-hotlist

@Katharine  any chance we can bump up the priority of this? It is a top source of oncall log spam

**Comment by BenTheElder** [comment]
is it possible to definitively identify this case based on `finished.json`? 
your sample link is now dead and I don't see any checking jobs currently in deck.

**Comment by fejta** [comment]
Are you using go/prow-debug? Happens continuously.

```yaml
{
 insertId: "1pbuwo1fvect0t"  
 jsonPayload: {
  component: "deck"   
  error: "failed to read started.json: failed to read pr-logs/pull/83583/pull-kubernetes-e2e-gce/1181653817947590656/started.json: failed to get reader for GCS object: storage: object doesn't exist"   
  file: "prow/cmd/deck/pr_history.go:143"   
  func: "main.getPRBuildData.func1"   
  level: "warning"   
  msg: "build pr-logs/pull/83583/pull-kubernetes-e2e-gce/1181653817947590656/ information incomplete"   
 }
 labels: {…}  
 logName: "projects/k8s-prow/logs/deck"  
 receiveTimestamp: "2019-10-09T20:33:36.262768634Z"  
 resource: {…}  
 severity: "ERROR"  
 timestamp: "2019-10-09T20:33:30Z"  
}
```

```yaml
{
 insertId: "1pbuwo1fvect0k"  
 jsonPayload: {
  component: "deck"   
  error: "failed to read started.json: failed to read pr-logs/pull/83583/pull-kubernetes-integration/1181653817964367873/started.json: failed to get reader for GCS object: storage: object doesn't exist"   
  file: "prow/cmd/deck/pr_history.go:143"   
  func: "main.getPRBuildData.func1"   
  level: "warning"   
  msg: "build pr-logs/pull/83583/pull-kubernetes-integration/1181653817964367873/ information incomplete"   
 }
 labels: {…}  
 logName: "projects/k8s-prow/logs/deck"  
 receiveTimestamp: "2019-10-09T20:33:36.262768634Z"  
 resource: {…}  
 severity: "ERROR"  
 timestamp: "2019-10-09T20:33:30Z"  
}
```

**Comment by BenTheElder** [comment]
I am, it doesn't actually happen continuously though... I couldn't find any when I looked for this, and during upgrades there are other errors.

**Comment by BenTheElder** [comment]
from the first entry above, it seems the results are by bootstrap.py and you can't detect this case from the finished.json but you could scan build-log.txt for `^.* merge: .* - not something we can merge$`

```
$ gsutil cat gs://kubernetes-jenkins/pr-logs/pull/83583/pull-kubernetes-e2e-gce/1181653817947590656/finished.json
{
  "timestamp": 1570563279, 
  "version": "unknown", 
  "result": "FAILURE", 
  "passed": false, 
  "job-version": "unknown", 
  "metadata": {
    "repo": "k8s.io/kubernetes", 
    "repos": {
      "k8s.io/kubernetes": "master:e856613dd5bb00bcfaca6974431151b5c06cbed5,83583:c5b6c7661918d5d0bd275380aacad485ad23b79d", 
      "k8s.io/release": "master"
    }, 
    "infra-commit": "f59e90136", 
    "pod": "652fa327-ea02-11e9-af79-6ac6ee30973f", 
    "repo-commit": "e856613dd5bb00bcfaca6974431151b5c06cbed5"
  }
}
```

alternatively, maybe we could test for the presence of started.json as a key and assume merge failed if it doesn't exist while finished.json does?

**Comment by fejta** [comment]
NOTE: Updated our stackdriver go link to ignore this. We should remove that once this closes

**Comment by Katharine** [comment]
The above is probably overcomplicating it - we just want to not explode when this information is missing. I don't think it's worth it to attempt to diagnose precisely why it is missing.

I think deck also, as it turns out, broadly does not care about the contents of this file. AFAICT the one place we do care we could learn the same information from `finished.json` regardless, and we should harden deck to do that. I think we should also make bootstrap actually consistently upload the information it is supposed to, given it seems unlikely to go away in the forseeable future.

(This also doesn't actually have anything to do with spyglass.)

Removing from the oncall hotlist now that the associated logs are filtered out.
/remove-kind oncall-hotlist

**Comment by Katharine** [comment]
/remove-lifecycle stale

This will be effectively resolved by #15785 or similar.
